packages/react/src/hooks/useStorageURL.ts

```diff
@@ -38,7 +40,12 @@ export const useStorageURL = (
     return () => Storage.cancel(promise);
   };
 
-  React.useEffect(fetch, [key, options, serializedOptions]);
+  React.useEffect(fetch, [key, serializedOptions]);
+
+  // Set the url to fallbackURL if error happens
+  if (result.error) {
+    result.url = fallbackURL;
+  }
```

Comment:
This should be done inside the catch block of `promise` to follow React conventions

Reply:
😄 See our discussion before, that will make `fallbackURL` a dependency in the effect <- https://github.com/aws-amplify/amplify-ui/pull/4054#discussion_r1231366005
But yeah, I do not have a strong opinion on this.